obsreport/obsreporttest/obsreporttest.go

```diff
@@ -77,7 +85,12 @@ func (tts *TestTelemetry) ToReceiverCreateSettings() component.ReceiverCreateSet
 // Shutdown unregisters any views and shuts down the SpanRecorder
 func (tts *TestTelemetry) Shutdown(ctx context.Context) error {
 	view.Unregister(tts.views...)
-	return tts.SpanRecorder.Shutdown(ctx)
+	var errors []error
+	errors = append(errors, tts.SpanRecorder.Shutdown(ctx))
+	if tts.meterProvider != nil {
+		errors = append(errors, tts.meterProvider.Shutdown(ctx))
+	}
+	return multierr.Combine(errors...)
```

Comment:
```suggestion
	var errs error
	errs = multierr.Append(errs, tts.SpanRecorder.Shutdown(ctx))
	if tts.meterProvider != nil {
		errs = multierr.Append(errs, tts.meterProvider.Shutdown(ctx))
	}
	return errs
```

This way avoid unnecessary slice allocations when only one error. Not performance critical, but good to do the correct/optimal thing every time since people will copy the code :))